### DISCUSSION TITLE
API Rule Problem "@request.auth.id = user.id"

### DISCUSSION THREAD

**Discussion by mgrami** [pocketbase/pocketbase#2702] [Q&A] [answered]
Hi. I'm using pockethost.io for hosting my pocketbase. It was good. I used "@request.auth.id = user.id" for my api rules so nobody can do anythings instead of others. It was working but recently I've notice that It doesn't let to do anything. when I left it empty it works.

**Comment by ganigeorgiev** [comment]
Are you sure that you are authenticated when sending the client request?
_If you are using the SDKs, for the test you can print `pb.authStore.isValid` before the request call._

What version of PocketBase do you use?

Could you provide an example how you are sending the request and eventually export your collections configuration as JSON (_Admin UI > Settings > Export collections_) so that I can try to reproduce it?

**Reply by ganigeorgiev** [reply]
Another thing that can fail the rule is if `user` is a multiple `relation` field (aka. it can have more than one user assigned). If that's the case either change it to single (_note that older PocketBase version doesn't update already inserted records_) or use the `?=` operator.

**Reply by mgrami** [reply]
First of all, thanks for the quick response.
Yes, I'm logged in. I'm using the sdk and it shows my username {$currentUser?.username}.
user is not a multiple relation field.
the weird thing is it was working. I don't know the version. I'm using pockethost.io for hosting my PB. Maybe they upgraded to a newer version and it caused the problem.

**Reply by mgrami** [reply]
You were right. this pb.authStore.isValid says false. But this {$currentUser?.username} says that I'm logged in !!! It writes my username.

export const pb = new PocketBase('https://pocket-maths.pockethost.io')
export const currentUser = writable(pb.authStore.model)

**Reply by ganigeorgiev** [reply, marked as answer]
It is possible that the token has expired. Keep in mind that `pb.authStore.model` is the last fetched state of your auth record and it doesn't guarantee that it is valid or that you are authenticated.

PocketBase is stateless, aka. there are no sessions in the traditional sense. A request is considered authenticated only when there is a valid non-expired `Authorization: TOKEN` header with the request, which is added automatically by the SDK with each request call.

**Reply by mgrami** [reply]
Now I understand what caused the problem. THANKS A LOT.
